### DISCUSSION TITLE
Is there a way to rename the automatically generated routes JSON files?

### DISCUSSION THREAD

**Discussion by fredrivett** [vercel/next.js#39923] [Help] [answered]
I have a problem, when I click to go to the `/analytics` page on my site, adblockers block the `analytics.json` file that's being requested by Next.js as they think it's an analytics tracker (it's not, it's a page listing analytics products).

Is there a way to rename the route files Next.js uses when navigating to server-side rendered pages on the client-side?

I want to either obfuscate the names so they're not machine readable, or have a way to rename them all.

Any help appreciated.

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/BW04O.png

**Comment by gaston-flores** [comment, marked as answer]
You can probably get around that using [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites).

Rename your `analytics.js` and set that as the destination for `/analytics`:

```js
// next.config.js
module.exports = {
  async rewrites() {
    return [
      {
        source: '/analytics',
        destination: '/nothing-suspicious-here', // will render nothing-suspicious-here.js
      },
    ]
  },
}
```

The browser will request `/nothing-suspicious-here` but the user will see `/analytics` as the path.

**Reply by fredrivett** [reply]
This is an ace idea, thanks so much @gaston-flores.

In my specific instance the `/analytics` page is a dynamic page category page, I've got over 30 of them, so it's structured like this: `pages/[category]/index.tsx`

I renamed the file to `pages/[category]/category.tsx`, and set up the rewrite like so:

```javascript
module.exports = {
  async rewrites() {
    return [
      {
        source: "/:category",
        destination: "/:category/category",
      },
    ];
  },
};
```

That was weirdly making the page URL `/analytics/category`, which might be due to having another partial match path in the same directory, `pages/[category]/[product].tsx`. I moved that to `pages/[category]/product/[product].tsx` and it seems to work as expected now.

Hopefully those extra details will help anyone else coming through to this.

Thanks again!
